### PR TITLE
[ENH] Extract gradient scheme from eddy output - for tractoflow 

### PR DESCRIPTION
**Pull request by arnaudbore**
The name is wrong I don't split. I average bvecs because when you run eddy you end up with not the right corresponding bvec bvals.

### CODE REVIEW

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-623/1/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-623/2/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-623/3/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-623/4/

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-623/6/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-623/8/

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

File: scripts/scil_validate_and_correct_eddy_gradients.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    assert_inputs_exist(parser, [args.in_bvec, args.in_bval])
+    assert_outputs_exist(parser, args, [args.out_bval, args.out_bvec])
+
+    """
+    IN BVEC
+    """
+    in_bvec = np.genfromtxt(args.in_bvec)
+    in_bvec_split = np.hsplit(in_bvec, int(in_bvec.shape[1]/args.nb_dirs))
+    if len(in_bvec_split)==2:
```
Comment (AlexVCaron) [inline scripts/scil_validate_and_correct_eddy_gradients.py:48]: If nb_dirs doesn't split the bvec array equally, we have a bug at the line after when averaging

File: scripts/scil_validate_and_correct_eddy_gradients.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter)
+    p.add_argument('in_bvec',
+                   help='In bvec file.')
+    p.add_argument('in_bval',
+                   help='In bval file.')
+    p.add_argument('nb_dirs', type=int,
+			       help='Number of directions per DWI.')
```
Comment (AlexVCaron) [inline scripts/scil_validate_and_correct_eddy_gradients.py:27]: Bad alignment

File: scripts/scil_validate_and_correct_eddy_gradients.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    assert_inputs_exist(parser, [args.in_bvec, args.in_bval])
+    assert_outputs_exist(parser, args, [args.out_bval, args.out_bvec])
+
+    """
+    IN BVEC
+    """
+    in_bvec = np.genfromtxt(args.in_bvec)
+    in_bvec_split = np.hsplit(in_bvec, int(in_bvec.shape[1]/args.nb_dirs))
+    if len(in_bvec_split)==2:
+        out_bvec = np.mean( np.array([ in_bvec_split[0], in_bvec_split[1] ]), axis=0 )
```
Comment (AlexVCaron) [inline scripts/scil_validate_and_correct_eddy_gradients.py:49]: Remove unnecessary whitespaces

File: scripts/tests/test_validate_and_correct_eddy_gradients.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+
+def test_execution_extract_half(script_runner):
+    os.chdir(os.path.expanduser(tmp_dir.name))
+    in_bvec = os.path.join(get_home(), 'processing', 'dwi.bvec')
+    in_bval = os.path.join(get_home(), 'processing', 'dwi.bval')
+    ret = script_runner.run('scil_validate_and_correct_eddy_gradients.py',
+                            in_bvec, in_bval, "32",
+                            'out.bvec',
+                            'out.bval')
```
Comment (AlexVCaron) [inline scripts/tests/test_validate_and_correct_eddy_gradients.py:26]: Aren't the tests shuffled at execution ? Add -f to this test case also if it ends up running before the one after

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
Some minor modifications. Good work !

File: scripts/tests/test_validate_and_correct_eddy_gradients.py (hunk 4 above)
Comment (arnaudbore) [reply scripts/tests/test_validate_and_correct_eddy_gradients.py:26]: Never seen it shuffled but I'm going to add the `-f`

**Review by AlexVCaron — APPROVED** [review]

**Comment by AlexVCaron** [comment]
After checks pass, LGTM !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Pull request merged**